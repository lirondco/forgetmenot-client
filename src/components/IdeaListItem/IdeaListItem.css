.IdeaListItem__main {
    background: #C03D72;
    margin: 3% 0 3% 0;
    width: 100%;
    height: fit-content;
    text-align: center;
    padding: 2%;
    border-radius: 4px;
}

.IdeaListItem__main a {
    text-decoration: none;
    font-family: 'Patrick Hand SC', cursive;
    color: #EFEBE9;
}

.delete_list_button {
    order: 1px solid #aaa;
    background-color: #FF6F00;
    border-radius: 4px;
}