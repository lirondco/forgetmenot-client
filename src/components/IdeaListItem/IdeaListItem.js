import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import IdeasApiService from '../../services/ideas-api-service'
import './IdeaListItem.css'

//this component renders each clickable list name

export default class IdeaListItem extends Component {
  static defaultProps = {
    onDeleteSuccess: () => {}
  }
  
  //function for deletion 
  handleClickDelete = e => {
    e.preventDefault()
    const listId = this.props.id
    if(window.confirm("WARNING: Deleting is irreversible and will delete all ideas in this list. Are you sure you want to delete?")) {
    IdeasApiService.deleteList(listId)
     .then(() => this.props.onDeleteSuccess())
     .catch(error=>console.error({error}))
  }}

  render() {
    const { list } = this.props
    return (
    <div className='IdeaListItem__main'>
      <Link to={`/lists/${list.id}`} className='IdeaListItem'>
        <header className='IdeaListItem__header'>
          <h2 className='IdeaListItem__heading'>
            {list.name}
          </h2>
        </header>
        <footer className='IdeaListItem__footer'>
          <ListIdeaCount list={list} />
        </footer>
      </Link>
      <button type='button' className='delete_list_button' onClick={this.handleClickDelete}>
          Delete
      </button>
    </div>
    )
  }
}

//function to display number of ideas in each list

function ListIdeaCount({ list }) {
  let text = "ideas";

  if(list.number_of_ideas < 2) {
    text = "idea"
  }
  return (
    <span
      className='IdeaListItem__idea-count'
    >
        {list.number_of_ideas} {text}
    </span>
  )
}