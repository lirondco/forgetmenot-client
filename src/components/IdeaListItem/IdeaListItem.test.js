import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import IdeaListItem from './IdeaListItem'

describe(`List component`, () => {
  const props = {
    list: {
        id: 3,
        name: 'testname',
        number_of_ideas: 4
    }
  }


  it('renders the list given props', () => {
    const wrapper = shallow(<IdeaListItem {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('button works when clicked', () => {
    const wrapper = shallow(<IdeaListItem {...props} />)
    wrapper.find('button').simulate('click', { preventDefault() {}})
})

})