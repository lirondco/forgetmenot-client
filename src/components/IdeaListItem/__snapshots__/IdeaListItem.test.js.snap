// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the list given props 1`] = `
<div
  className="IdeaListItem__main"
>
  <Link
    className="IdeaListItem"
    to="/lists/3"
  >
    <header
      className="IdeaListItem__header"
    >
      <h2
        className="IdeaListItem__heading"
      >
        testname
      </h2>
    </header>
    <footer
      className="IdeaListItem__footer"
    >
      <ListIdeaCount
        list={
          Object {
            "id": 3,
            "name": "testname",
            "number_of_ideas": 4,
          }
        }
      />
    </footer>
  </Link>
  <button
    className="delete_list_button"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</div>
`;
